server/odc-service/src/main/java/com/oceanbase/odc/service/connection/ConnectionService.java

```diff
@@ -423,9 +423,19 @@ public PaginatedData<ConnectionConfig> listByProjectId(@NotNull Long projectId,
         if (basic) {
             connections = repository.findByDatabaseProjectId(projectId).stream().map(e -> {
                 ConnectionConfig c = new ConnectionConfig();
+                Environment environment = environmentService.detail(e.getEnvironmentId());
```

Comment:
On complex, we need to avoid this~

Reply:
yes, avoid MetaDB access in loop, which may result in bad performance

Reply:
I don't quite understand. Do you mean that the method of executing `.detail` in `stream` is complicated?

Reply:
I don't quite understand. Do you mean that the method of executing `.detail` in `stream` is complicated?

> yes, avoid MetaDB access in loop, which may result in bad performance

got it. I just checked the implementation and found this problem.

Reply:
I don't quite understand. Do you mean that the method of executing `.detail` in `stream` is complicated?

> yes, avoid MetaDB access in loop, which may result in bad performance

got it. I just checked the implementation and found this problem.

> On complex, we need to avoid this~

Yes, I guessed it. Thanks for the reminder.